core_sign_update

```diff
@@ -136,7 +136,8 @@ i=1
 signature_sizes=""
 for key in "${private_keys[@]}"; do
 	if [[ "${key}" == pkcs11* ]]; then
-		openssl rsautl -engine pkcs11 -pkcs -sign -inkey ${key} -keyform engine -in update.pkcs11-padhash -out update.sig.${i}
+		# NOTE: When we will use OpenSSL 3: use the following URI: pkcs11:id=%${ID}?pin-source=file:/tmp/pin
+		openssl pkeyutl -engine pkcs11 -sign -keyform engine -inkey "pkcs11:id=%${ID}?pin-value=$(cat /tmp/pin)" -in update.pkcs11-padhash -out update.sig.2
```

Comment:
Why does it write to `update.sig.2` instead of `update.sig.${i}`?

Reply:
And `${key}` is the PKCS#11 URI where I think the ID and pin-value should be part of. So instead of using cat /tmp this has to be specified in the invocation through the private_keys arg.
```suggestion
		openssl pkeyutl -engine pkcs11 -sign -keyform engine -inkey "${key}" -in update.pkcs11-padhash -out update.sig.${i}
```

Reply:
In flatcar-build-scripts/generate_payload we pass a hardcoded `pkcs11:object=10;type=private` but we can totally change that to make use of a variable populated from a `read` to enter the pin.

Reply:
(while at it we should delete the `dummy.key.pem` cargo cult)

Reply:
> (while at it we should delete the `dummy.key.pem` cargo cult)

needs omitting or using 2 for `--public_key_version` in `core_sign_update`